modules/es-cluster-health-is-red.adoc

```diff
@@ -0,0 +1,153 @@
+----
+
+. If there are pending tasks, monitor their status. If their status changes and indicates that the cluster is recovering, continue waiting. The recovery time varies according to the size of the cluster and other factors. Otherwise, if the status of the pending tasks does not change, this indicates that the recovery has stalled.
+
+. If it seems like the recovery has stalled, check if the `cluster.routing.allocation.enable` value is set to `none`, by running the following command and observing the output:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cluster/settings?pretty
+----
+
+. If the `cluster.routing.allocation.enable` value is set to `none`, set it to `all`, by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cluster/settings?pretty -X PUT -d '{"persistent": {"cluster.routing.allocation.enable":"all"}}'
+----
+
+. Check if any indices are still red by running the following command and observing the output:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cat/indices?v
+----
+
+. If any indices are still red, try to clear them by performing the following steps.
+
+.. Clear the cache by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_index_name>/_cache/clear?pretty
+----
+
+.. Increase the max allocation retries by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_index_name>/_settings?pretty -X PUT -d '{"index.allocation.max_retries":10}'
+----
+
+.. Delete all the scroll items by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_search/scroll/_all -X DELETE
+----
+
+.. Increase the timeout by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_index_name>/_settings?pretty -X PUT -d '{"index.unassigned.node_left.delayed_timeout":"10m"}'
+----
+
+. If the preceding steps do not clear the red indices, delete the indices individually.
+
+.. Identify the red index name by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cat/indices?v
+----
+
+.. Delete the red index by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_red_index_name> -X DELETE
```

Comment:
```suggestion
$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query="<elasticsearch_red_index_name> -X DELETE"
```

Reply:
Per https://redhat-internal.slack.com/archives/C02321AKYTT/p1696866637850469?thread_ts=1696524477.252959&cid=C02321AKYTT we will leave this as is